src/test/java/io/strimzi/kafka/bridge/http/DisablingConsumerProducerIT.java

```diff
@@ -0,0 +1,186 @@
+import io.strimzi.kafka.bridge.config.KafkaProducerConfig;
+import io.strimzi.kafka.bridge.http.model.HttpBridgeError;
+import io.strimzi.kafka.bridge.http.services.ConsumerService;
+import io.strimzi.kafka.bridge.http.services.ProducerService;
+import io.strimzi.kafka.bridge.utils.Urls;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static io.strimzi.kafka.bridge.Constants.HTTP_BRIDGE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@ExtendWith(VertxExtension.class)
+@Tag(HTTP_BRIDGE)
+public class DisablingConsumerProducerIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DisablingConsumerProducerIT.class);
+
+    private static Vertx vertx;
+    private static WebClient client;
+
+    private static final Map<String, Object> config = new HashMap<>();
+    private static final String BRIDGE_EXTERNAL_ENV = System.getenv().getOrDefault("EXTERNAL_BRIDGE", "FALSE");
+    private static final int TEST_TIMEOUT = 600000000;
+
+    static {
+        // NOTE: despite other tests, these not need the Kafka container because the requests should not need it to be successful
+
+        long timeout = 5L;
+        config.put(KafkaConfig.KAFKA_CONFIG_PREFIX + ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        config.put(KafkaConsumerConfig.KAFKA_CONSUMER_CONFIG_PREFIX + ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        config.put(KafkaProducerConfig.KAFKA_PRODUCER_CONFIG_PREFIX + ProducerConfig.MAX_BLOCK_MS_CONFIG, "10000");
+        config.put(HttpConfig.HTTP_CONSUMER_TIMEOUT, timeout);
+        config.put(BridgeConfig.BRIDGE_ID, "my-bridge");
+    }
+
+    @BeforeEach
+    void setup(VertxTestContext context) {
+        vertx = Vertx.vertx();
+        client = WebClient.create(vertx, new WebClientOptions()
+                .setDefaultHost(Urls.BRIDGE_HOST)
+                .setDefaultPort(Urls.BRIDGE_PORT)
+        );
+        context.completeNow();
```

Comment:
I am not vertx expert, but are you sure this won't make all tests pass immediately even if the test fails? I remember we hit some confusing behavior in the operator repo in the past because of this.

Also wouldn't be better to use pure Java for new things?

Reply:
That was my understanding but it's not the case, if you see all the other tests in the bridge.

> Also wouldn't be better to use pure Java for new things?

We are going to move out from Vert.x so the tests will be refactored again. I would leave these ones as Vert.x for now, then there will be an entire work to rewrite them without the Vert.x support.